app/services/email_normalizer.rb

```diff
@@ -33,6 +35,8 @@ def google_mx_record?
   end
 
   def mx_records(domain)
+    domain = Addressable::IDNA.to_ascii(domain)
```

Comment:
I don't like re-assigning variables, can we rename one or the other?

```suggestion
  def mx_records(unescaped_domain)
    domain = Addressable::IDNA.to_ascii(unescaped_domain)
```

Or rename elsewhere
```suggestion
  def mx_records(domain)
    ascii_domain = Addressable::IDNA.to_ascii(domain)
```